pkg/services/ngalert/ngalert.go

```diff
@@ -164,15 +164,42 @@ func (ng *AlertNG) init() error {
 	// If enabled, configure the remote Alertmanager.
 	// - If several toggles are enabled, the order of precedence is RemoteOnly, RemotePrimary, RemoteSecondary
 	// - If no toggles are enabled, we default to using only the internal Alertmanager
-	// We currently support only remote secondary mode, so in case other toggles are enabled we fall back to remote secondary.
+	// We currently do not support remote primary mode, so we fall back to remote secondary.
 	var overrides []notifier.Option
 	moaLogger := log.New("ngalert.multiorg.alertmanager")
 	remoteOnly := ng.FeatureToggles.IsEnabled(initCtx, featuremgmt.FlagAlertmanagerRemoteOnly)
 	remotePrimary := ng.FeatureToggles.IsEnabled(initCtx, featuremgmt.FlagAlertmanagerRemotePrimary)
 	remoteSecondary := ng.FeatureToggles.IsEnabled(initCtx, featuremgmt.FlagAlertmanagerRemoteSecondary)
 	if ng.Cfg.UnifiedAlerting.RemoteAlertmanager.Enable {
 		switch {
-		case remoteOnly, remotePrimary:
+		case remoteOnly:
+			ng.Log.Debug("Starting Grafana with remote only mode enabled")
+			m := ng.Metrics.GetRemoteAlertmanagerMetrics()
+			m.Info.WithLabelValues(metrics.ModeRemoteSecondary).Set(1)
+
+			// This function will be used by the MOA to create new Alertmanagers.
+			override := notifier.WithAlertmanagerOverride(func(factoryFn notifier.OrgAlertmanagerFactory) notifier.OrgAlertmanagerFactory {
+				return func(ctx context.Context, orgID int64) (notifier.Alertmanager, error) {
+					// Create remote Alertmanager.
+					cfg := remote.AlertmanagerConfig{
+						BasicAuthPassword: ng.Cfg.UnifiedAlerting.RemoteAlertmanager.Password,
+						DefaultConfig:     ng.Cfg.UnifiedAlerting.DefaultConfiguration,
+						OrgID:             orgID,
+						TenantID:          ng.Cfg.UnifiedAlerting.RemoteAlertmanager.TenantID,
+						URL:               ng.Cfg.UnifiedAlerting.RemoteAlertmanager.URL,
```

Comment:
You're missing the `Promoted` field. This way we'll add `promoted: true` to the payload when we send a configuration, indicating to the remote AM that it should use it.

```suggestion
						URL:               ng.Cfg.UnifiedAlerting.RemoteAlertmanager.URL,
						PromoteConfig: true,
```